metadata-ingestion/src/datahub/ingestion/source/s3/source.py

```diff
@@ -819,7 +820,10 @@ def local_browser(self, path_spec: PathSpec) -> Iterable[Tuple[str, datetime, in
                 dirs.sort(key=functools.cmp_to_key(partitioned_folder_comparator))
 
                 for file in sorted(files):
-                    full_path = os.path.join(root, file)
+                    # We need to make sure the path is in posix style which is not true on windows
+                    full_path = PurePath(
+                        os.path.normpath(os.path.join(root, file))
+                    ).as_posix()
```

Comment:
Do we need both os.path.normpath and PurePath.as_posix ?

From docs, both do similar things.
https://docs.python.org/3/library/os.path.html#os.path.normpath
https://docs.python.org/3/library/pathlib.html

Reply:
If I get it right doesn't do the same thing even though there are similarities:
1. normpath -> 
```
os.path.normpath() method in Python is used to normalize the specified path. All redundant separator and up-level references are collapsed in the process of path normalization. 

For example: A//B, A/B/, A/./B and A/foo/../B all will be normalized to A/B. 

On Windows operating system, any forward slash (‘/’) in the path is converted to backslash (‘\’).
```
This is needed to make the dir separator consistent and also it simplifies the path (which I think is a good thing)
2. PurePath's as_posix turns the path to a UNIX path and uses unix type dir separators which is what s3 source and the urn expects

Reply:
right they aren't same. Sounds like normpath does something more than as_posix, but yet it already does everything that as_posix might be doing ? (linux like  / separators)

Reply:
Didn't understand the value add of PurePath.as_posix over os.path.normpath but the change seems better than original code.